src/winbind.c

```diff
@@ -9,13 +9,74 @@
 
 #include <wbclient.h>
 
+#ifdef HAVE_PTHREAD
+#include <pthread.h>
+
+static pthread_key_t key;
+static pthread_once_t key_once = PTHREAD_ONCE_INIT;
+
+/* not that the destructor is given directly the content of the key in
+ * ptr, not the key itself. */
+static void key_destructor(void *ptr)
+{
+    wbcCtxFree((struct wbcContext *)ptr);
+}
+
+static void key_create(void)
+{
+    /* we have no way to report errors ... */
+    (void)pthread_key_create(&key, key_destructor);
```

Comment:
Shouldn't we abort here?  Failure will garbage up the context value; if we're lucky, the pthread_setspecific() call will fail with EINVAL:

       If an implementation detects that the value specified by the key argu‐
       ment to pthread_setspecific() does not refer to a a key value obtained
       from pthread_key_create() or refers to a key that has been deleted
       with pthread_key_delete(), it is recommended that the function should
       fail and report an [EINVAL] error.

Reply:
I rather just have gssntlmp fail to operate than abort a user program